frontend/src/lib/components/TaxonomicFilter/cohortFilterUtils.ts

```diff
@@ -0,0 +1,43 @@
+import { BehavioralFilterKey } from 'scenes/cohorts/CohortFilters/types'
+
+import { AnyCohortCriteriaType, CohortCriteriaGroupFilter, CohortType } from '~/types'
+
+function isCohortCriteriaGroupFilter(
+    value: AnyCohortCriteriaType | CohortCriteriaGroupFilter
+): value is CohortCriteriaGroupFilter {
+    return (value as CohortCriteriaGroupFilter).type === 'AND' || (value as CohortCriteriaGroupFilter).type === 'OR'
+}
+
+const hasBehavioralFilter = (cohort: CohortType, allCohorts: CohortType[]): boolean => {
+    const checkCriteriaGroup = (group: CohortCriteriaGroupFilter): boolean => {
```

Comment:
very nice! For a follow up, check if this slows down the list display too much, since we're calculating this for every group in every cohort, and traversing the chain every time. It _shouldn't_ be bad, but I'm not sure when we get up to 10k cohorts and have an O(n^2) traversal. Should be pretty alright for most teams though!

It looks like there's no pagination on cohorts model as well, so we always load all cohorts, which works well for now ;) (just keep in mind if team-product-analytics for example introduces pagination on that endpoint, we've done that before for a few endpoints and broken random things 😅 )